src/groupBy.ts

```diff
@@ -54,13 +54,13 @@ const _groupBy = (indexed: boolean) => <T>(
 };
 
 export namespace groupBy {
-  export function indexed<T, K>(
-    array: readonly T[],
-    fn: PredIndexed<T, PropertyKey>
-  ): Record<string, NonEmptyArray<T>>;
-  export function indexed<T, K>(
-    fn: PredIndexed<T, PropertyKey>
-  ): (array: readonly T[]) => Record<string, NonEmptyArray<T>>;
+  export function indexed<Item, Key extends PropertyKey>(
+    array: readonly Item[],
+    fn: PredIndexed<Item, Key>
+  ): Record<Key, NonEmptyArray<Item>>;
```

Comment:
I'm not using the indexed version, but shouldn't it have same return type?

Reply:
thx for seeing that, you are right 🙏 

b66c4167